lms/djangoapps/discussion/rest_api/permissions.py

```diff
@@ -185,3 +185,41 @@ def has_permission(self, request, view):
             request.user.is_staff or
             is_user_staff and request.method == "GET"
         )
+
+
+class IsAllowedToBulkDelete(permissions.BasePermission):
+    """
+    Permission that checks if the user is staff or an admin.
+    """
+
+    def has_permission(self, request, view):
+        """Returns true if the user can bulk delete posts"""
+        if not request.user.is_authenticated:
+            return False
+
+        course_ids = CourseEnrollment.objects.filter(user=request.user).values_list('course_id', flat=True)
+
+        user_roles = set(
+            Role.objects.filter(
+                users=request.user,
+                course_id__in=course_ids,
+            ).values_list('name', flat=True).distinct()
+        )
+        is_user_staff = bool(user_roles & {
+            FORUM_ROLE_ADMINISTRATOR,
+            FORUM_ROLE_MODERATOR,
+        })
+        if is_user_staff is True:
+            return True
+
+        for course_id in course_ids:
+            if (
+                CourseStaffRole(course_id).has_user(request.user) or
```

Comment:
This check can be optimized by fetching all active user course staff and course instructor roles. fetch only course_ids then  comparing them with the enrollment list